### PR TITLE
feat: do not send any histogram or summary metric

### PR DESCRIPTION
**Pull request by sumo-drosiek**
### Checklist

Histogram is a lot of data points which are not used on Sumo Logic side by default. In case we will need to send some of them, we can figure out additional flag which will be an exception from that rule, but for now it sounds reasonable to stop ingesting buckets at all

The same is about summary and quantiles

- [ ] Changelog updated or skip changelog label added
- [ ] Documentation updated
- [ ] Template tests added for new features
- [ ] Integration tests added or modified for major features

### CODE REVIEW

**Review by kkujawa-sumo — APPROVED** [review]

**Comment by kkujawa-sumo** [comment]
Please remember to add changelog entry.

**Comment by sumo-drosiek** [comment]
Not quite sure about that, but we may probably also want to do the same for summary metrics

File: tests/integration/helm_prometheus_metrics_test.go [hunk 1]
```diff
@@ -14,7 +14,7 @@ func Test_Helm_Prometheus_Metrics(t *testing.T) {
 	// defaults without otel metrics collector metrics, but with Prometheus metrics
 	expectedMetricsGroups := make([][]string, len(internal.DefaultExpectedMetricsGroups))
 	copy(expectedMetricsGroups, internal.DefaultExpectedMetricsGroups)
-	expectedMetricsGroups = append(expectedMetricsGroups, internal.PrometheusMetrics, internal.DefaultOtelcolMetrics)
+	expectedMetricsGroups = append(expectedMetricsGroups, internal.PrometheusMetrics, internal.DefaultOtelcolMetrics, internal.BucketMetrics)
```
Comment (sumo-drosiek) [inline tests/integration/helm_prometheus_metrics_test.go:17]: I'm not convinced to this change. as we filter out all buckets in metadata layer, it shouldn't be needed. Maybe telegraf receiver do not create histogram metrics 🤔

**Review by rnishtala-sumo — APPROVED** [review]

File: deploy/helm/sumologic/conf/metrics/otelcol/processors.yaml [hunk 2]
```diff
@@ -16,7 +16,7 @@ filter/drop_unnecessary_metrics:
       - resource.attributes["job"] != "pod-annotations" and IsMatch(name, "scrape_.*")
 {{- if .Values.sumologic.metrics.dropHistogramBuckets }}
       # drop histograms we've extracted sums and counts from, but don't want the full thing
-      - IsMatch(name, "^(apiserver_request_duration_seconds|coredns_dns_request_duration_seconds|kubelet_runtime_operations_duration_seconds)$")
+      - type == METRIC_DATA_TYPE_HISTOGRAM or type == METRIC_DATA_TYPE_EXPONENTIAL_HISTOGRAM or type == METRIC_DATA_TYPE_SUMMARY or IsMatch(name, ".*_bucket")
```
Comment (rnishtala-sumo) [inline deploy/helm/sumologic/conf/metrics/otelcol/processors.yaml:19]: how would we send one or two histogram metrics?

File: deploy/helm/sumologic/conf/metrics/otelcol/processors.yaml [hunk 3]
```diff
@@ -16,7 +16,7 @@ filter/drop_unnecessary_metrics:
       - resource.attributes["job"] != "pod-annotations" and IsMatch(name, "scrape_.*")
```
Comment (rnishtala-sumo) [inline deploy/helm/sumologic/conf/metrics/otelcol/processors.yaml:16]: should we also drop metrics coming in from pod annotations by default?

File: deploy/helm/sumologic/conf/metrics/otelcol/processors.yaml (hunk 2 above)
Comment (sumo-drosiek) [reply deploy/helm/sumologic/conf/metrics/otelcol/processors.yaml:19]: by exposing new configuration option, and adding `not isMatch` at the end of the query

File: deploy/helm/sumologic/conf/metrics/otelcol/processors.yaml (hunk 3 above)
Comment (sumo-drosiek) [reply deploy/helm/sumologic/conf/metrics/otelcol/processors.yaml:16]: It's breaking change

File: deploy/helm/sumologic/conf/metrics/otelcol/processors.yaml (hunk 2 above)
Comment (sumo-drosiek) [reply deploy/helm/sumologic/conf/metrics/otelcol/processors.yaml:19]: https://github.com/SumoLogic/sumologic-kubernetes-collection/issues/3819

Comment (sumo-drosiek) [reply deploy/helm/sumologic/conf/metrics/otelcol/processors.yaml:19]: I wouldn't play with `merge` if we can do it easily by the configuration option

**Pull request merged**